Monitor for file changes inside MAKD_TMPDIR, not inside /tmp

nemanja-boric-sociomantic opened the pull request:
Travis environment doesn't report file changes/events for files created
inside /tmp directory via inotify, so this creates the files & monitors
them inside MAKD_TMPDIR, if set.

mathias-lang-sociomantic commented on test/filesystemevent/main.d:
```diff
@@ -154,19 +158,26 @@ class FileCreationTestTask: Task
 
     ***************************************************************************/
-
+import ocean.io.Stdout;
```
Leftover ? (Or unordered)

nemanja-boric-sociomantic replied:
No, debug commit.

nemanja-boric-sociomantic committed:
Don't verify the path is registered inside FileSystemEvent.handle

Similar to all ISelectClient which allow modifying the state inside its
`handle` method, we can unregister the path while handling the other
event inside FileSystemEvent, and the event for now unregistered path
was already raised (so it must be handled), so we just want to ignore
this event, not to throw.

nemanja-boric-sociomantic committed:
Unwatch the directory after the fileCreationTest has completed

We don't need to monitor for the directory anymore after we confirmed
monitoring the create files work. This makes it easier to create the
file for the next tests inside that directory, not raising IN_CREATE event.

nemanja-boric-sociomantic committed:
Monitor for file changes inside MAKD_TMP, not inside /tmp

Travis environment doesn't report file changes/events for files created
inside /tmp directory via inotify, so this creates the files & monitors
them inside MAKD_TMPDIR, if set.

nemanja-boric-sociomantic committed:
Switch the FileSystemEvent test to use TaskEvent

Instead of manually checking if the task is suspended, just use
TaskEvent to wait/trigger the test synchronization points.

nemanja-boric-sociomantic commented:
OK, ready for review/merge. :green_heart:

nemanja-boric-sociomantic commented:
I'm going to revisit `DirectorySandbox` to use `MAKD_TMPPATH` if available by default in v4.x.x

mihails-strasuns-sociomantic commented on test/filesystemevent/main.d:
```diff
@@ -113,6 +113,7 @@ class FileCreationTestTask: Task
         this.suspend();
 
         theScheduler.epoll.unregister(inotifier);
+        inotifier.unwatch(this.watched_path.dup);
```
Looks like inotifier was not very nicely const-ported :)

nemanja-boric-sociomantic replied:
Yeah, I'll take a look now we got this green how to improve it.

mihails-strasuns-sociomantic reviewed (approved):
<3 Thank you very much!
